src/endpoints/customers.js

```diff
@@ -5,33 +5,44 @@ class CustomersEndpoint extends CRUDExtend {
     super(endpoint)
 
     this.endpoint = 'customers'
+
+    this.sendToken = (tokenRequestBody, headers = {}) =>
+      this.request.send(
+        `${this.endpoint}/tokens`,
+        'POST',
+        tokenRequestBody,
+        null,
+        {
+          ...headers
+        }
+      )
+  }
+
+  TokenViaPassword(email, password, headers) {
+    const body = {
+      type: 'token',
+      authentication_mechanism: 'password',
+      email,
+      password
+    }
+
+    return this.sendToken(body, headers)
+  }
+
+  TokenViaOIDC(code, redirectUri, codeVerifier, headers) {
+    const body = {
+      type: 'token',
+      authentication_mechanism: 'oidc',
+      oauth_authorization_code: code,
+      oauth_redirect_uri: redirectUri,
+      oauth_code_verifier: codeVerifier
+    }
+
+    return this.sendToken(body, headers)
   }
 
-  Token(email, password, code, redirectUri, headers) {
-    const tokenRequestBody = (() => {
-      const body = {
-        type: 'token'
-      }
-      if (code && redirectUri) {
-        body.oauth_authorization_code = code
-        body.oauth_redirect_uri = redirectUri
-        body.authentication_mechanism = 'oidc'
-      } else {
-        body.email = email
-        body.password = password
-      }
-      return body
-    })()
-
-    return this.request.send(
-      `${this.endpoint}/tokens`,
-      'POST',
-      tokenRequestBody,
-      null,
-      {
-        ...headers
-      }
-    )
+  Token(email, password) {
```

Comment:
What is the point of having both `.Token()` and `.TokenViaPassword()`? This can be confusing to customers. We have to keep `.Token()` for backward compatibility so I think we should just keep `.Token()` and `.TokenViaOIDC()`.

Reply:
> What is the point of having both .Token() and .TokenViaPassword()?

The point is to ensure API source compatibility and not break existing clients. We need to have some kind of API deprecation strategy, and that typically starts with renaming methods, and keeping the old methods around for some time. 

For examples in JavaScript, the [Regexp Compile](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/compile).

> This can be confusing to customers.

Yes I 100% agree this will confuse **some customers**. I also 100% think that it will confuse **some customers** if we have a Token method that doesn't take all the parameters or options that the Token endpoint does. If we were designing this API from scratch, we wouldn't have a `Token()` and `TokenViaOIDC()` method with the existing signatures.

>We have to keep .Token() for backward compatibility so I think we should just keep .Token() and .TokenViaOIDC().

If our API policy is that from now until the sun expands, we _never_ remove a method, and we never rename a method then sure I agree with that. Java is known for prioritizing long term backwards compatibility and never removes methods, even dangerously unsafe ones, but they do create better and more intentional methods. 

What we [discussed in Slack](https://elasticpath-software.slack.com/archives/CB6811UJJ/p1603993835065700?thread_ts=1603983569.058000&cid=CB6811UJJ) is that we would add these two methods, and Token (which I did in the other file), and then we could remove Token at a later date.

So in summary, the point is that this is a standard technique that people use to evolve APIs while maintaining source compatibility. I am happy if we don't want this, but @samblacklock  said yes, so we should circle back with him if that isn't what we want.

Reply:
@shaunmaharaj what are your thoughts on the above?